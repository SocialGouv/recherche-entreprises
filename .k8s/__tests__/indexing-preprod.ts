//

import { getEnvManifests } from "@socialgouv/kosko-charts/testing";
import { project } from "@socialgouv/kosko-charts/testing/fake/github-actions.env";

jest.setTimeout(1000 * 60);
test("kosko generate --dev", async () => {
  expect(
    await getEnvManifests("preprod", "jobs/indexing", {
      ...project("recherche-entreprises").preprod,
      RANCHER_PROJECT_ID: "c-bd7z2:p-wbfnt",
    })
  ).toMatchSnapshot();
});
