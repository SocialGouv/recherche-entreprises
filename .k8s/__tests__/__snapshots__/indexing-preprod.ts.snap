// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kosko generate --dev 1`] = `
"---
apiVersion: bitnami.com/v1alpha1
kind: SealedSecret
metadata:
  name: elastic-recherche-entreprises-write
  annotations:
    app.gitlab.com/app: socialgouv-recherche-entreprises
    app.gitlab.com/env: preprod-dev42
    app.gitlab.com/env.name: preprod-dev42
  labels:
    application: v1-2-3-recherche-entreprises
    component: v1-2-3-recherche-entreprises
    owner: recherche-entreprises
    team: recherche-entreprises
    cert: wildcard
  namespace: recherche-entreprises-85-preprod-dev42
spec:
  encryptedData:
    ELASTICSEARCH_URL: >-
      AgB82d7LnDA/6RxRlks/7cRtmWuz6Ql1p4gCwdghu3X85Ek4FqSL6ui1tIBiuM1pZcaiwM6ZizsKA0ofq5iBafUwRQOzTFc3XAMy7XltrymG/QwBRmYKS4w4Ub1DPuYpVxEUC6Jngyex7OvhCKUK7pugjG6Q8FXO6i9iyVVEpKnAcSVLaUe+olmlOrO2RMjIK3mgKX/xOFT+2FYiN5/LJob+w/+p0hPlZaMsLrLOl/i5N4LuI5ckg+FawifD2MnN057fsLbwt0m63g7ZHvXtGT66tbTcQgpWfy5kLe2m7oIbzdk+oPoh4FS8PnyU6nMC8sOkC3v/GUMK91qCas01RBoyPRTTs11yX3gbYHti5Nc3zDt36YHPhrqfRQHQ8xONYkx5SkAylnDr1JoXyfrKDwZUBvLQ6Xh3gGI7qu909LxZ2ryWd9WRslpB1+8bOLN0tV20slAesGYFC/W6e5GT0AhwWqwJ9usGLf0dM7GE+IXJegIlconcM+2x/FW3RQ5XnK5kI/coiha5pxBkK0p8pbwmLnOwH5c2QoBD5xgLCZ3wleMcbTWdSynzm/LSYkWZzL15M3dy4m8c+qXc88LCbAHTZ2UaAH/XF7pcuMvOF33ZesjgaWLumFoUvhtaG1gZN97eU2/K1xLwo+x/vt06P6vUbU2Emj9cEziTaQqx1ZPUI4Hsp5ZNpbAGRsLMhCf7G7YvxZkpxh+7GAKvW1JmhF1DYgBapCen2bxQN/XK/pNniL89T0hiKSwh/Fo8yDOnqAGova3T1Nq6fLFcRYNxtKFVlsn5zsUgyVPCtyn+cUKIFw==
    ELASTICSEARCH_API_KEY: >-
      AgCWc+7OMggEJuNUXxL8CufezAMvJaT3svLkFi9KiKost/avFQNBoXDGixl7vERoCDVvDxnPlcn7m7MIjveP5oIfewhJzYQua9WmQrSMTab1soqUIjMHWj+2A8y0qSYg4s21w/X6bc+H0/O4+ax0QNVtm/MWc30vnCOF+uVibM2WcDkSY/FGE2bE7hcmQWeDbsmnRxDaZTY10ME2ycZpv9eAMoXshCQ98k4LFT9DM0D51az4BibLJjUtsW/vMn/FNKv9/teOGiCAFE/pfoyeZ4QN7ZsoFzCSHwj++Rd0hWScX8VCvbS5pgmfioz8SHmQFrxrN994CAPlr2Rw7mxfkzSETrDgzQVmzKD+j49PZL6cz77cCh9DGE70Pco91szWbsaUQ/lcWUTlFm39Z0Xmf6uV1eCSbyPcM3ogMx95Rjb3MnGa2zrt4OPLMMA2YfyqMtrrVTcq2aJ4fGvWAmVZHH3k2QMcPiYLGHQ8gnVsjI6LYGJEM27RVj99/ppZpoPWeims00fBHMclr3/4czfQCBKcr8GFouxfcKVYGJ4gRNdH0Qn5gZIPXOOiRKUVS0ik0zuT1Xu9u9kU6Df2RVOQawiIOy14+5RrMqt0li1PuCFZjRTqbsLGogOzJQfaXKOafNjxswUqXSn4gR3pVBmm0tPJx+9iFdSrkavvGBrQeSDrNzY5UjeZwIgz9hveUUe1dxARBK8iXu0Q8P6W+VnFActyQs0gnoc4Zzb3jJNfxgDWv1LF/MK22b5G2YcZ3bk5LSsGxP1KICJ0J1MMSL8=
  template:
    metadata:
      annotations:
        app.gitlab.com/app: socialgouv-recherche-entreprises
        app.gitlab.com/env: preprod-dev42
        app.gitlab.com/env.name: preprod-dev42
      name: elastic-recherche-entreprises-write
      labels:
        application: v1-2-3-recherche-entreprises
        component: v1-2-3-recherche-entreprises
        owner: recherche-entreprises
        team: recherche-entreprises
        cert: wildcard
    type: Opaque
---
apiVersion: v1
kind: ConfigMap
data:
  sqlite.sh: >
    #!/bin/bash


    # download files, convert to SQLite and export to CSV


    DATA_DIR=\${DATA_DIR:-\\"./data\\"}


    mkdir -p \\"$DATA_DIR\\" || true


    echo \\"-- Working in $(dirname \\"$0\\")\\"

    cd \\"$(dirname \\"$0\\")\\" || exit


    echo \\"-- Download datasets\\"


    # install sqlite3 if not exists

    if ! command -v sqlite3 &> /dev/null

    then
        echo \\"sqlite3 could not be found\\"
        apt-get update -y
        apt-get install -y sqlite3
    fi

    # install wget if not exists

    if ! command -v wget &> /dev/null

    then
        echo \\"wget could not be found\\"
        apt-get update -y
        apt-get install -y wget
    fi

    # install unzip if not exists

    if ! command -v unzip &> /dev/null

    then
        echo \\"unzip could not be found\\"
        apt-get update -y
        apt-get install -y unzip
    fi


    for d in $(seq -w 1 19) 2A 2B $(seq 21 74) $(seq 76 95) 98 \\"\\"; do
      wget --progress=bar:force:noscroll -q --show-progress \\"https://files.data.gouv.fr/geo-sirene/last/dep/geo_siret_$d.csv.gz\\" --directory-prefix=\\"$DATA_DIR\\"
      gunzip \\"\${DATA_DIR}/geo_siret_$d.csv.gz\\"
    done


    #Cas particulier Paris

    for d in $(seq -w 1 20); do
      wget --progress=bar:force:noscroll -q --show-progress \\"https://files.data.gouv.fr/geo-sirene/last/dep/geo_siret_751$d.csv.gz\\" --directory-prefix=\\"$DATA_DIR\\"
      gunzip \\"\${DATA_DIR}/geo_siret_751$d.csv.gz\\"
    done


    #Cas particulier DOM

    for d in $(seq -w 1 8); do
      wget --progress=bar:force:noscroll -q --show-progress \\"https://files.data.gouv.fr/geo-sirene/last/dep/geo_siret_97$d.csv.gz\\" --directory-prefix=\\"$DATA_DIR\\"
      gunzip \\"\${DATA_DIR}/geo_siret_97$d.csv.gz\\"
    done


    # SIRET data

    wget --progress=bar:force:noscroll -q --show-progress
    https://files.data.gouv.fr/insee-sirene/StockUniteLegale_utf8.zip
    --directory-prefix=\\"$DATA_DIR\\"

    unzip \\"\${DATA_DIR}/StockUniteLegale_utf8.zip\\" -d \\"\${DATA_DIR}\\"


    # WEEZ data

    wget --progress=bar:force:noscroll -q --show-progress
    https://www.data.gouv.fr/fr/datasets/r/a785345a-6e8c-4961-ae0a-bc00878e4f2e
    -O \\"\${DATA_DIR}/WEEZ.csv\\"


    echo \\"-- Import CSV datasets to sqlite\\"


    sqlite3 -echo db.sqlite \\".read import.sql\\"


    echo \\"-- Export sqlite data to CSV\\"


    sqlite3 -header -csv db.sqlite \\".read export.sql\\" > output.csv
  import.sql: |-
    --- sqlite3 -echo db.sqlite \\".read import-sqlite.sql\\"
    ---
    --- import local CSVs to sqlite database
    ---

    PRAGMA synchronous = OFF;

    DROP TABLE IF EXISTS weez;
    DROP TABLE IF EXISTS geo_siret;
    DROP TABLE IF EXISTS stock;

    .mode csv
    .import data/WEEZ.csv weez
    .import data/StockUniteLegale_utf8.csv stock
    .import data/geo_siret_.csv geo_siret
    .import data/geo_siret_01.csv geo_siret
    .import data/geo_siret_02.csv geo_siret
    .import data/geo_siret_03.csv geo_siret
    .import data/geo_siret_04.csv geo_siret
    .import data/geo_siret_05.csv geo_siret
    .import data/geo_siret_06.csv geo_siret
    .import data/geo_siret_07.csv geo_siret
    .import data/geo_siret_08.csv geo_siret
    .import data/geo_siret_09.csv geo_siret
    .import data/geo_siret_10.csv geo_siret
    .import data/geo_siret_11.csv geo_siret
    .import data/geo_siret_12.csv geo_siret
    .import data/geo_siret_13.csv geo_siret
    .import data/geo_siret_14.csv geo_siret
    .import data/geo_siret_15.csv geo_siret
    .import data/geo_siret_16.csv geo_siret
    .import data/geo_siret_17.csv geo_siret
    .import data/geo_siret_18.csv geo_siret
    .import data/geo_siret_19.csv geo_siret
    .import data/geo_siret_21.csv geo_siret
    .import data/geo_siret_22.csv geo_siret
    .import data/geo_siret_23.csv geo_siret
    .import data/geo_siret_24.csv geo_siret
    .import data/geo_siret_25.csv geo_siret
    .import data/geo_siret_26.csv geo_siret
    .import data/geo_siret_27.csv geo_siret
    .import data/geo_siret_28.csv geo_siret
    .import data/geo_siret_29.csv geo_siret
    .import data/geo_siret_2A.csv geo_siret
    .import data/geo_siret_2B.csv geo_siret
    .import data/geo_siret_30.csv geo_siret
    .import data/geo_siret_31.csv geo_siret
    .import data/geo_siret_32.csv geo_siret
    .import data/geo_siret_33.csv geo_siret
    .import data/geo_siret_34.csv geo_siret
    .import data/geo_siret_35.csv geo_siret
    .import data/geo_siret_36.csv geo_siret
    .import data/geo_siret_37.csv geo_siret
    .import data/geo_siret_38.csv geo_siret
    .import data/geo_siret_39.csv geo_siret
    .import data/geo_siret_40.csv geo_siret
    .import data/geo_siret_41.csv geo_siret
    .import data/geo_siret_42.csv geo_siret
    .import data/geo_siret_43.csv geo_siret
    .import data/geo_siret_44.csv geo_siret
    .import data/geo_siret_45.csv geo_siret
    .import data/geo_siret_46.csv geo_siret
    .import data/geo_siret_47.csv geo_siret
    .import data/geo_siret_48.csv geo_siret
    .import data/geo_siret_49.csv geo_siret
    .import data/geo_siret_50.csv geo_siret
    .import data/geo_siret_51.csv geo_siret
    .import data/geo_siret_52.csv geo_siret
    .import data/geo_siret_53.csv geo_siret
    .import data/geo_siret_54.csv geo_siret
    .import data/geo_siret_55.csv geo_siret
    .import data/geo_siret_56.csv geo_siret
    .import data/geo_siret_57.csv geo_siret
    .import data/geo_siret_58.csv geo_siret
    .import data/geo_siret_59.csv geo_siret
    .import data/geo_siret_60.csv geo_siret
    .import data/geo_siret_61.csv geo_siret
    .import data/geo_siret_62.csv geo_siret
    .import data/geo_siret_63.csv geo_siret
    .import data/geo_siret_64.csv geo_siret
    .import data/geo_siret_65.csv geo_siret
    .import data/geo_siret_66.csv geo_siret
    .import data/geo_siret_67.csv geo_siret
    .import data/geo_siret_68.csv geo_siret
    .import data/geo_siret_69.csv geo_siret
    .import data/geo_siret_70.csv geo_siret
    .import data/geo_siret_71.csv geo_siret
    .import data/geo_siret_72.csv geo_siret
    .import data/geo_siret_73.csv geo_siret
    .import data/geo_siret_74.csv geo_siret
    .import data/geo_siret_75101.csv geo_siret
    .import data/geo_siret_75102.csv geo_siret
    .import data/geo_siret_75103.csv geo_siret
    .import data/geo_siret_75104.csv geo_siret
    .import data/geo_siret_75105.csv geo_siret
    .import data/geo_siret_75106.csv geo_siret
    .import data/geo_siret_75107.csv geo_siret
    .import data/geo_siret_75108.csv geo_siret
    .import data/geo_siret_75109.csv geo_siret
    .import data/geo_siret_75110.csv geo_siret
    .import data/geo_siret_75111.csv geo_siret
    .import data/geo_siret_75112.csv geo_siret
    .import data/geo_siret_75113.csv geo_siret
    .import data/geo_siret_75114.csv geo_siret
    .import data/geo_siret_75115.csv geo_siret
    .import data/geo_siret_75116.csv geo_siret
    .import data/geo_siret_75117.csv geo_siret
    .import data/geo_siret_75118.csv geo_siret
    .import data/geo_siret_75119.csv geo_siret
    .import data/geo_siret_75120.csv geo_siret
    .import data/geo_siret_76.csv geo_siret
    .import data/geo_siret_77.csv geo_siret
    .import data/geo_siret_78.csv geo_siret
    .import data/geo_siret_79.csv geo_siret
    .import data/geo_siret_80.csv geo_siret
    .import data/geo_siret_81.csv geo_siret
    .import data/geo_siret_82.csv geo_siret
    .import data/geo_siret_83.csv geo_siret
    .import data/geo_siret_84.csv geo_siret
    .import data/geo_siret_85.csv geo_siret
    .import data/geo_siret_86.csv geo_siret
    .import data/geo_siret_87.csv geo_siret
    .import data/geo_siret_88.csv geo_siret
    .import data/geo_siret_89.csv geo_siret
    .import data/geo_siret_90.csv geo_siret
    .import data/geo_siret_91.csv geo_siret
    .import data/geo_siret_92.csv geo_siret
    .import data/geo_siret_93.csv geo_siret
    .import data/geo_siret_94.csv geo_siret
    .import data/geo_siret_95.csv geo_siret
    .import data/geo_siret_971.csv geo_siret
    .import data/geo_siret_972.csv geo_siret
    .import data/geo_siret_973.csv geo_siret
    .import data/geo_siret_974.csv geo_siret
    .import data/geo_siret_975.csv geo_siret
    .import data/geo_siret_976.csv geo_siret
    .import data/geo_siret_977.csv geo_siret
    .import data/geo_siret_978.csv geo_siret
    .import data/geo_siret_98.csv geo_siret

    CREATE INDEX 'geo_siret_idx' ON 'geo_siret' ('siret');
    CREATE INDEX 'geo_siren_idx' ON 'geo_siret' ('siren');
    CREATE INDEX 'weez_siret_idx' ON 'weez' ('SIRET');
    CREATE INDEX 'stock_siren_idx' ON 'stock' ('siren');

    SELECT \\"weez\\", count(*) from weez;
    SELECT \\"stock\\", count(*) from stock;
    SELECT \\"geo_siret\\", count(*) from geo_siret;
  export.sql: |+
    --- Output index data
    --- sqlite3 -header -csv db.sqlite \\".read export.sql\\" > output.csv

    PRAGMA synchronous = OFF;

    SELECT 
        stock.siren,
        stock.sigleUniteLegale,
        stock.nomUniteLegale,
        stock.nomUsageUniteLegale,
        stock.denominationUniteLegale,
        stock.denominationUsuelle1UniteLegale,
        stock.denominationUsuelle2UniteLegale,
        stock.denominationUsuelle3UniteLegale,
        stock.activitePrincipaleUniteLegale,
        stock.trancheEffectifsUniteLegale,
        stock.categorieJuridiqueUniteLegale,
        stock.nomenclatureActivitePrincipaleUniteLegale,
        stock.categorieEntreprise,
        stock.etatAdministratifUniteLegale,
        stock.caractereEmployeurUniteLegale,
        geo_siret.siret,
        geo_siret.codePostalEtablissement,
        geo_siret.libelleCommuneEtablissement,
        geo_siret.etatAdministratifEtablissement,
        geo_siret.enseigne1Etablissement,
        geo_siret.enseigne2Etablissement,
        geo_siret.enseigne3Etablissement,
        geo_siret.denominationUsuelleEtablissement,
        geo_siret.activitePrincipaleEtablissement,
        geo_siret.geo_adresse,
        weez.IDCC as idcc,
        (select count(*) FROM geo_siret where siren=stock.siren) etablissements
        from stock, geo_siret
            left join weez on weez.SIRET=geo_siret.siret
            where stock.siren=geo_siret.siren;


metadata:
  name: config-map-files-0123456
  annotations:
    app.gitlab.com/app: socialgouv-recherche-entreprises
    app.gitlab.com/env: preprod-dev42
    app.gitlab.com/env.name: preprod-dev42
  labels:
    application: v1-2-3-recherche-entreprises
    component: v1-2-3-recherche-entreprises
    owner: recherche-entreprises
    team: recherche-entreprises
    cert: wildcard
  namespace: recherche-entreprises-85-preprod-dev42
---
apiVersion: batch/v1
kind: Job
metadata:
  name: update-index-0123456
  annotations:
    app.gitlab.com/app: socialgouv-recherche-entreprises
    app.gitlab.com/env: preprod-dev42
    app.gitlab.com/env.name: preprod-dev42
  labels:
    application: v1-2-3-recherche-entreprises
    component: v1-2-3-recherche-entreprises
    owner: recherche-entreprises
    team: recherche-entreprises
    cert: wildcard
  namespace: recherche-entreprises-85-preprod-dev42
spec:
  backoffLimit: 3
  template:
    spec:
      containers:
        - name: update-index
          image: >-
            harbor.fabrique.social.gouv.fr/cdtn/recherche-entreprises-index:1.2.3
          volumeMounts:
            - name: data
              mountPath: /data
          env:
            - name: ASSEMBLY_FILE
              value: /data/output.csv
          envFrom:
            - secretRef:
                name: elastic-recherche-entreprises-write
          resources:
            limits:
              cpu: '4'
              memory: 5Gi
            requests:
              cpu: '2'
              memory: 2Gi
      restartPolicy: Never
      volumes:
        - name: data
          emptyDir: {}
        - configMap:
            name: config-map-files-0123456
            defaultMode: 511
          name: local-files
      initContainers:
        - args:
            - /mnt/scripts/sqlite.sh
          command:
            - sh
          image: ubuntu:18.04
          imagePullPolicy: Always
          name: download-build-data
          volumeMounts:
            - name: data
              mountPath: /data
            - mountPath: /mnt/scripts
              name: local-files
    metadata:
      annotations:
        app.gitlab.com/app: socialgouv-recherche-entreprises
        app.gitlab.com/env: preprod-dev42
        app.gitlab.com/env.name: preprod-dev42
      labels:
        application: v1-2-3-recherche-entreprises
        component: v1-2-3-recherche-entreprises
        owner: recherche-entreprises
        team: recherche-entreprises
        cert: wildcard
"
`;
